### ISSUE TITLE
cacheManager.getRuntimeConfiguration().getClassloader() returns null

### ISSUE THREAD

**Issue by mathieucarbou** [ehcache/ehcache3#1160]
Don't know whether it's a bug or doc issue (to me it's a bug, getClassloader() should always return I think from a user perspective - this is what I would think it does).

But now it returns null. The javadoc says: 

"The ClassLoader must not be null."

**Comment by mathieucarbou** [comment]
RuntimeConfiguration has 2 implementations.

`XmlConfiguration` defaults to `ClassLoading.getDefaultClassLoader()` and even has a null-check:

```
if(classLoader == null) {
      throw new NullPointerException("The classLoader can not be null");
    }
```

So I propose to change `DefaultConfiguration` from:

```
  public DefaultConfiguration(Map<String, CacheConfiguration<?, ?>> caches, ClassLoader classLoader, ServiceCreationConfiguration<?>... services) {
    this.services = unmodifiableCollection(Arrays.asList(services));
    this.caches = new ConcurrentHashMap<String, CacheConfiguration<?, ?>>(caches);
    this.classLoader = classLoader;
  }
```

to

```
  public DefaultConfiguration(Map<String, CacheConfiguration<?, ?>> caches, ClassLoader classLoader, ServiceCreationConfiguration<?>... services) {
    this.services = unmodifiableCollection(Arrays.asList(services));
    this.caches = new ConcurrentHashMap<String, CacheConfiguration<?, ?>>(caches);
    this.classLoader = classLoader == null ? ClassLoading.getDefaultClassLoader() : classLoader;
  }
```

**Comment by ljacomet** [comment]
Re-opening as this needs to be fixed on `release/3.0` as well.
